### ISSUE TITLE
W030 false positive

### ISSUE THREAD

**Issue by VincentGuinaudeau** [jshint/jshint#2717]
Hello,

I've juste install JShint on atom and after some configuration it's working pretty well.

But here is a case where the errors does't seem to come from me :

```javascript
	if (this.type === OBJECT_TYPE) {
		this.location[tab[1]] = location
		++this.max // W030 : Expected an assignment or function call and instead saw an expression
		return true
	}
```

By putting the ++ operand at the end of the statement or a semi-colon on the line above the warning deapper but both syntax are valid and increment this.max.

**Comment by nicolo-ribaudo** [comment]
You can workaround this by adding a semicolon after `location` (or before `++`, as you prefer)

**Comment by VincentGuinaudeau** [comment]
yes, i've notice i can workaround this, but since it's a assigment and JShint say the opposite, i thinked this as to be reported.

**Comment by nicolo-ribaudo** [comment]
On master that code also throws `Missing semicolon.` (`E058`) between `++` and `this`

**Comment by nicolo-ribaudo** [comment]
I made some attempts to fix this issue, but I don't know which is the best :sweat_smile:

1. nicolo-ribaudo/jshint@9cd07e2475008212bdd5c1799b93cc8b6868394b

  This is probably the simpler method. If `++` is on a new line, the `expression` function stops looping. I don't like this because I feel like the `expression` function should know as least as possible about the JavaScript syntax.

2. nicolo-ribaudo/jshint@be0af7de93a47aff9ed237b00f7877a0e1b1e997

  Disable the `led` if `++` is on a new line (prefix operators should just have the `nud` method)
